### PR TITLE
Ceate account service CRUD and ban endpoint

### PR DESCRIPTION
**Pull request by AlB1111122**
## Summary
https://trello.com/c/dXJuOhr8
 - create user service module in the java project
 - modify structure to allow it to be made up of multipule modules
 - add create, update, and delete endpoints
 - add a ban endpoint
 - the user services db and the user table is automatically built on docker compose up

## How to verify
./gradlew build
docker build -t springbootdemo . (build the project aplication in that was already there)
docker build -t userservice UserService  (build the userservice)
test out the endpoints

### CODE REVIEW

File: UserService/src/main/java/CS4337/Project/User.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+  public String getUsername() {
+    return this.username;
+  }
+
+  public String getEmail() {
+    return this.email;
+  }
+
+  public String getAdress() {
```
Comment (oOoOoOo0o) [inline UserService/src/main/java/CS4337/Project/User.java:59]: Small typo here, same with setAdress

File: UserService/src/main/java/CS4337/Project/UserService.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@SpringBootApplication
+@RestController
+public class UserService {
+
+  @Autowired private JdbcTemplate jdbcTemplate;
+
+  // @Autowired private SimpleJdbcInsert insert;
```
Comment (oOoOoOo0o) [inline UserService/src/main/java/CS4337/Project/UserService.java:28]: Should this comment be removed?

File: UserService/src/main/java/CS4337/Project/UserService.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+  public static void main(String[] args) {
+    SpringApplication.run(UserService.class, args);
+  }
+
+  @GetMapping("/")
+  public Map<String, Object> sayhello() {
```
Comment (oOoOoOo0o) [inline UserService/src/main/java/CS4337/Project/UserService.java:35]: Is this endpoint necessary?

File: UserService/src/main/java/CS4337/Project/UserService.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        updateStatement.append("address = ?, ");
+        params.add(userUpdates.getAdress());
+      }
+
+      updateStatement.setLength(updateStatement.length() - 2); // Remove last ", "
+      updateStatement.append(" WHERE id = ?");
+      params.add(id);
+
+      int updatedRows = jdbcTemplate.update(updateStatement.toString(), params.toArray());
+      return ResponseEntity.ok(Map.of("sucess", String.valueOf(updatedRows)));
```
Comment (oOoOoOo0o) [inline UserService/src/main/java/CS4337/Project/UserService.java:121]: Typo in "sucess", should be success

**Review by oOoOoOo0o — CHANGES_REQUESTED** [review]
Only small things, looks good otherwise

**Review by oOoOoOo0o — APPROVED** [review]

**Review by tadhgryan2012 — APPROVED** [review]
Looks good to me!

**Review by s1m0nWasTaken — APPROVED** [review]
Looks good , thanks.

**Comment by AlB1111122** [comment]
[create account service](https://trello.com/c/dXJuOhr8/16-create-account-service)

**Pull request merged**